iamy/aws.go

```diff
@@ -383,6 +398,11 @@ func (a *AwsFetcher) getAccount() (*Account, error) {
 	return &acct, nil
 }
 
+// Wrapper around getAccount method to make mocking easier
+var getAccount = func (a *AwsFetcher) (*Account, error) {
+	return a.getAccount()
+}
```

Comment:
Using Globals to facilitate mocking isn't a very idiomatic way to do this.
How I'd approach it is to create an interface
```golang
type AwsFetcherIface interface {
	GetAccount() (*Account, error)
	FetchIAMData() error
	FetchS3Data() error
}
```

Then instantiate AwsFetcher and pass it into the function being tested. That way we can define our own implementation in the test suite and pass it in to the function.

Reply:
Got it! Will probably steal your work in https://github.com/99designs/iamy/pull/58 - I tried to do it but didn't grok interfaces well and ended up with a very long declaration.

Reply:
Struggling a little - the function under test is a method of AwsFetcher as well (AwsFetcher's `Fetch()`, which is used by push/pull, will call `getAccount()`, `fetchIamData()` and `fetchS3Data` methods). My limited understanding is that in creating an interface for AwsFetcher, I would also be mocking Fetch, which makes the test meaningless.

Do you have any advice? I'm deliberating:
- creating interfaces for S3/IAM clients and mocking those before passing into Fetcher - it's quite tedious though, there's potentially a lot to mock when all I want to do is check if fetchS3Data / fetchIamData is being called
- creating a separate type to hold the 3 methods and embedding that into Fetcher (something like [this](https://stackoverflow.com/a/42839979/1839532)), but it feels a little contrived

Reply:
I think breaking it up is a great idea, but also would like @mtibben to weigh in.

Reply:
👌